### ISSUE TITLE
CQL2: Remove `-` from list of valid identifier characters?

### ISSUE THREAD

**Issue by jerstlouis** [opengeospatial/ogcapi-features#766]
It conflicts with the minus arithmetic operator.
With the current rules, `a-b` is interpreted as identifier `"a-b"` instead of `a` minus `b`, and needs to be re-written as `a - b` with spaces to mean `a` minus `b`.
This is unexpected and very confusing.
Identifiers actually containing hyphens could still be escaped as `"a-b"`.

e.g., http://5.9.22.176:8080/ogcapi/collections/montreal/items?filter=maxLat-minLat>0.01%20and%20maxLon-minLon>0.01
vs.
http://5.9.22.176:8080/ogcapi/collections/montreal/items?filter=maxLat%20-%20minLat>0.01%20and%20maxLon%20-%20minLon>0.01

(CMSS) http://5.9.22.176:8080/ogcapi/collections/montreal/items?filter=maxLat-minLat>0.01%20and%20maxLon-minLon>0.01&filter-lang=cmss

**Comment by cportele** [comment]
Meeting 2022-11-21: @pvretano will remove the dash from identifierPart, and also add text that explains which chars can start an identifier and which can be used in the rest of the identifier in plain English. After that he will merge the PR.

**Comment by pvretano** [comment]
PR #753 has been updated to remove the dash from identifier start and add some English words describing the valid characters that can appear in an identifier.
